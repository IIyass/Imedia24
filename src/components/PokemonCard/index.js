import React from "react";
import styled from "styled-components";
import { string } from "prop-types";

const PokemonCardContainer = styled.div`
  cursor: pointer;
  border: 2px solid #2a75bb;
  display:flex;
  h1 {
    color: #ffcb05;
    margin:auto;
    padding:20px;
  }
  :hover {
    transition: all 0.3s ease 0s;
    transform: translateY(-10px);
    transition-duration: 500ms;
    box-shadow: rgba(0, 0, 0, 0.2) 0px 1px 5px 5px,
      rgba(0, 0, 0, 0.14) 0px 2px 2px 0px, rgba(0, 0, 0, 0.12) 0px 3px 1px -2px;
  }
`;
const PokemonCard = ({ name }) => (
  <PokemonCardContainer>
    <h1>{name}</h1>
  </PokemonCardContainer>
);


PokemonCard.propTypes = {
  name: string,
};

export default PokemonCard;
