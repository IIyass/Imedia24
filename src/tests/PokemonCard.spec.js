import React from 'react';
import { shallow ,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PokemonCard from '../components/PokemonCard/index';

configure({adapter: new Adapter()});

describe("PokemonCard", () => {
  it("should render my pokemon card", () => {
    const component = shallow(<PokemonCard name="pikatchu"/>);
    expect(component.getElements()).toMatchSnapshot();
  });
});