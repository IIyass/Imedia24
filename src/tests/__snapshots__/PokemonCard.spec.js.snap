// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should render my pokemon card 1`] = `
Array [
  <styled.div>
    <h1>
      pikatchu
    </h1>
  </styled.div>,
]
`;
